spec/integ/crypto/verification.spec.ts

```diff
@@ -1221,6 +1523,64 @@ function expectSendToDeviceMessage(msgtype: string): Promise<{ messages: any }>
     });
 }
 
+/**
+ * Utility to add all needed mocks for secret requesting (to device of type `m.secret.request`).
+ *
+ * The following secrets are mocked: `m.cross_signing.master`, `m.cross_signing.self_signing`,
+ * `m.cross_signing.user_signing`, `m.megolm_backup.v1`.
+ *
+ * @returns a map of secret name to promise that will resolve when the secret is requested.
+ */
+function mockUpSecretRequestAndGetPromises(): Map<string, Promise<string>> {
+    let mskRequested: (requestId: string) => void;
+    const awaitMskRequest: Promise<string> = new Promise((resolve) => {
+        mskRequested = resolve;
+    });
```

Comment:
this is fine, but for future reference, consider something like:

```typescript
    let mskRequestDefer = defer<string>();
    promiseMap.set("m.cross_signing.master", mskRequestDefer.promise);
    
    // ...

    if (name == "m.cross_signing.master") {
        mskRequestDefer.resolve(requestId);
    }
   
```

(`defer` from `matrix-js-sdk/src/utils.ts`). And then

Reply:
Thanks for the tip, a lot better.